menu.py

```diff
@@ -1,38 +1,452 @@
+            menu_keys.append("shift")
+        if modifiers & kMenuOptionModifier:
+            menu_keys.append("alt")
+        if modifiers & kMenuControlModifier:
+            menu_keys.append("ctrl")
+        if modifiers & kMenuFnGlobeModifier:
+            menu_keys.append("fn")
+
+    got_key = False
+
+    if key_char is not None:
+        menu_keys.append(key_char.lower())
+        got_key = True
+
+    if not got_key and virtual_key is not None:
+        if key_name := VK_NAMES.get(virtual_key):
+            menu_keys.append(key_name)
+            got_key = True
+
+    # XXX(nriley) consider accounting for glyphs in some cases
+    # Example: "Num Lock" in Terminal has kVK_Escape but kMenuClearGlyph
+    # However, cmd-esc works fine to trigger it
+
+    if not got_key:
+        no_key_message = []
+        if virtual_key is not None:
+            no_key_message.append(f"virtual key code {virtual_key:X}")
+        if glyph is not None:
+            no_key_message.append(f"glyph {glyph:X}")
+        if no_key_message:
+            print("Unsupported key with", ", ".join(no_key_message))
+            print(selected_menu.dump())
+            app.notify("Key not supported", "\n".join(no_key_message))
+        else:
+            app.notify("Key not found")
+        return None, None, None
+
+    return "-".join(menu_keys), menu_path, strategy
+
+
 @mod.action_class
 class Actions:
     def copy_menu_select():
         """Copies TalonScript to select the menu item that is currently highlighted"""
-        selected_menu = active_menubar()
-        menu_path = []
-        while True:
-            if not selected_menu.AXSelectedChildren:
-                break
-            selected_item = selected_menu.AXSelectedChildren[0]
-            menu_path.append(
-                selected_item.AXTitle.replace("\\", r"\\").replace("|", r"\|")
-            )
-            # XXX(nriley) this is True even if list empty - report Talon bug? children also doesn't support len()
-            # print(selected_item.children, bool(selected_item.children))
-            try:
-                selected_menu = selected_item.children.find_one(
-                    AXRole="AXMenu", max_depth=0
-                )
-            except ui.UIErr:
-                break
+        _, menu_path, strategy = selected_menu_path_strategy()
+
+        if not menu_path:
+            app.notify("No menu bar item selected or under the mouse pointer")
+            return
+
+        escaped_menu_path = [
+            title.replace("\\", r"\\").replace("|", r"\|") for title in menu_path
+        ]
+
+        clip.set_text(f'user.menu_select({"|".join(escaped_menu_path)!r})')
+        app.notify(
+            "Copied TalonScript to select menu item",
+            body=f'{" ▸ ".join(menu_path)}\n{strategy}',
+        )
+
+    def copy_menu_key():
+        """Copies TalonScript to press the key equivalent of the menu item that is currently highlighted"""
+
```

Comment:
```suggestion
        """Copies TalonScript to press the key equivalent of the menu item that is currently highlighted"""
```

Reply:
Was this a mistake? I can't tell the difference between the original version and your suggested change.

Reply:
Oh, sorry, I was suggesting to delete the empty line afterwards. I probably should have included the following line in the suggestion context.

Reply:
Gotcha. Yes, I should be consistent here — removed the blank line in this and the following function.